pkg/apis/pipeline/v1/matrix.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2023 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
```

Comment:
suggest bringing the type [definition](https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1/pipeline_types.go#L207) here 🙏

Reply:
I moved the definition to matrix_types.go